src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -601,11 +601,16 @@ MavlinkMissionManager::handle_mission_ack(const mavlink_message_t *msg)
 	if (CHECK_SYSID_COMPID_MISSION(wpa)) {
 		if ((msg->sysid == _transfer_partner_sysid && msg->compid == _transfer_partner_compid)) {
 			if (_state == MAVLINK_WPM_STATE_SENDLIST && _mission_type == wpa.mission_type) {
+
 				_time_last_recv = hrt_absolute_time();
 
-				if (_transfer_seq == current_item_count()) {
+				if (wpa.type == MAV_MISSION_ACCEPTED && _transfer_seq == current_item_count()) {
```

Comment:
This is probably minor, but I'm wondering if there's any benefit in separately handling the mission_ack type check and the mission sync state (eg _transfer_seq).

For example if wpa.type isn't `MAV_MISSION_ACCEPTED` or `MAV_MISSION_OPERATION_CANCELLED` do we want to handle that error separately (debug logging) and also stop the transfer (`_transfer_in_progress = false`).

Reply:
We are eventually logging all cases in the `else` below. Generally, I try to look into these cases when they happen one by one.

Reply:
This is really minor, but isn't the message we send to the ground (send_statustext_critical) on failure now potentially wrong? The transfer could finish (all items) and if the client populated the wrong MAV_MISSION_RESULT we'd send a bogus error.

Reply:
Agreed, if a GCS sends a NACK, we should not send an error.

Reply:
Ok, I removed the `send_statustext_critical` in 0141ab3cd3.